kwasm-lemmas.md

```diff
@@ -325,8 +347,15 @@ They are non-trivial in their implementation, but the following should obviously
       requires WIDTH ==Int 1
       [simplification]
 
-    rule #getRange(BM, ADDR, WIDTH) modInt 256 => #get(BM, ADDR)
-      requires notBool (WIDTH ==Int 0)
+    rule #getRange(BM, ADDR, WIDTH) modInt BYTE_SIZE => #get(BM, ADDR)
+      requires BYTE_SIZE ==Int 256
```

Comment:
I don't think the haskell backend gets any benefit having this be in the side condition and not just directly in the rule.

The java backend would use the rule version to generate `==K`, but the haskell backend will specialize it to `==Int` when it sees that it is integers on both sides of the `==K`.

Reply:
There are situations where the `BYTE_SIZE` doesn't exactly match 256, but may instead be `X *Int 8` with `X ==Int 32` in a side condition, in which case this rule would fail to apply, since it's a `[simplification]`, right? Because the LHS must exactly *match* for the rule to apply.
https://github.com/kframework/k/blob/master/pending-documentation.md#simplification-attribute-haskell-backend

Or have I misunderstood what "match" means in this case? I think I've seen examples of this, where a simplification didn't apply. The simplification was:
```
#substrBytes(BS, 0, END) => BS requires lengthbytes(BS) ==Int END [simplification]
```
but the actual expression in the `<k>` cell was 
```
#substrBytes(ADDRESS, lengthBytes(SELECTOR) -Int 4, 20)
```
with `lengthBytes(SELECTOR) ==Int 4` in a side condition, or something like that.